pkg/config/config.go

```diff
@@ -250,6 +272,58 @@ func (mpp *MountPodPatch) merge(mp MountPodPatch) {
 	if mp.TerminationGracePeriodSeconds != nil {
 		mpp.TerminationGracePeriodSeconds = mp.TerminationGracePeriodSeconds
 	}
+	vok := make(map[string]bool)
+	if mp.Volumes != nil {
+		if mpp.Volumes == nil {
+			mpp.Volumes = []corev1.Volume{}
+		}
+		for _, v := range mp.Volumes {
+			if IsInterVolume(v.Name) {
+				klog.Warningf("applyConfig: volume %s uses an internal volume name, ignore", v.Name)
+				continue
+			}
+			found := false
+			for _, vv := range mpp.Volumes {
+				if vv.Name == v.Name {
+					found = true
+					break
+				}
+			}
+			if found {
+				klog.Warningf("applyConfig: volume %s already exists, ignore", v.Name)
+				continue
+			}
+			vok[v.Name] = true
+			mpp.Volumes = append(mpp.Volumes, v)
+		}
+	}
+	if mp.VolumeMounts != nil {
+		if mpp.VolumeMounts == nil {
+			mpp.VolumeMounts = []corev1.VolumeMount{}
+		}
+		for _, vm := range mp.VolumeMounts {
```

Comment:
Does `volumeMounts` also need to check interVolume?

Reply:
no need, it will check the exist of the used volume, which has already been check interVolume